cmds/gzip/gzip.go

```diff
@@ -0,0 +1,436 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/klauspost/pgzip"
+	"github.com/spf13/pflag"
+)
+
+const version string = "0.0.1"
+
+type options struct {
+	blocksize  int
+	level      int
+	processes  int
+	decompress bool
+	force      bool
+	help       bool
+	keep       bool
+	license    bool
+	list       bool
+	name       bool
+	noName     bool
+	quiet      bool
+	recursive  bool
+	rsyncable  bool
+	stdout     bool
+	test       bool
+	verbose    bool
+	version    bool
+	suffix     string
+}
+
+func (o *options) parseArgs() error {
+
+	levels := make([]bool, 10)
+
+	pflag.IntVarP(&o.blocksize,
+		"blocksize",
+		"b",
+		128,
+		"Set compression block size in KiB")
+
+	pflag.BoolVarP(&o.decompress,
+		"decompress",
+		"d",
+		false,
+		"Decompress the compressed input")
+
+	pflag.BoolVarP(&o.force,
+		"force",
+		"f",
+		false,
+		"Force overwrite of output file and compress links")
+
+	pflag.BoolVarP(&o.help,
+		"help",
+		"h",
+		false,
+		"Display a help screen and quit")
+
+	pflag.BoolVarP(&o.keep,
+		"keep",
+		"k",
+		false,
+		"Do not delete original file after processing")
+
+	pflag.BoolVarP(&o.license,
+		"license",
+		"L",
+		false,
+		"Display license")
+
+	pflag.BoolVarP(&o.list,
+		"list",
+		"l",
+		false,
+		"List the contents of the compressed input [NOT IMPLEMENTED]")
+
+	pflag.BoolVarP(&o.name,
+		"name",
+		"N",
+		false,
+		"Store/restore file name and mod time in/from header [NOT IMPLEMENTED]")
+
+	pflag.BoolVarP(&o.noName,
+		"no-name",
+		"n",
+		false,
+		"Do not store or restore file name in/from header [NOT IMPLEMENTED]")
+
+	pflag.IntVarP(&o.processes,
+		"processes",
+		"p",
+		runtime.NumCPU(),
+		"Allow up to n compression threads")
+
+	pflag.BoolVarP(&o.quiet,
+		"quiet",
+		"q",
+		false,
+		"Print no messages, even on error [NOT IMPLEMENTED]")
+
+	pflag.BoolVarP(&o.recursive,
+		"recursive",
+		"r",
+		false,
+		"Process the contents of all subdirectories [NOT IMPLEMENTED]")
+
+	pflag.BoolVarP(&o.rsyncable,
+		"rsyncable",
+		"R",
+		false,
+		"Input-determined block locations for rsync [NOT IMPLEMENTED]")
+
+	pflag.BoolVarP(&o.stdout,
+		"stdout",
+		"c",
+		false,
+		"Write all processed output to stdout (won't delete)")
+
+	pflag.StringVarP(&o.suffix,
+		"suffix",
+		"S",
+		".gz",
+		"Specify suffix for compression")
+
+	pflag.BoolVarP(&o.test,
+		"test",
+		"t",
+		false,
+		"Test the integrity of the compressed input [NOT IMPLEMENTED]")
```

Comment:
Let's not include options that aren't implemented?

Reply:
Another option is we hide the flags from usage output until their implemented. Only reason I can think of to keep them as valid flags and hide them is if we need the CLI compatibility somewhere.

Reply:
Leaning towards just commenting out the unimplemented flags and option struct attributes for now though.